server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateFinder.java

```diff
@@ -44,19 +44,16 @@ public QualityGateFinder(DbClient dbClient) {
    * Return effective quality gate of a project.
    *
    * It will first try to get the quality gate explicitly defined on a project, if none it will try to return default quality gate.
-   * As it's possible to have no default quality gate, this method can return {@link Optional#empty()}
    */
-  public Optional<QualityGateData> getQualityGate(DbSession dbSession, long componentId) {
+  public QualityGateData getQualityGate(DbSession dbSession, OrganizationDto organization, long componentId) {
     Optional<Long> qualityGateId = dbClient.projectQgateAssociationDao().selectQGateIdByComponentId(dbSession, componentId);
     if (qualityGateId.isPresent()) {
       QualityGateDto qualityGate = checkFound(dbClient.qualityGateDao().selectById(dbSession, qualityGateId.get()), "No quality gate has been found for id %s", qualityGateId);
-      return Optional.of(new QualityGateData(qualityGate, false));
+      return new QualityGateData(qualityGate, false);
     } else {
-      Optional<QualityGateDto> defaultQualityGate = getDefault(dbSession);
-      if (!defaultQualityGate.isPresent()) {
-        return Optional.empty();
-      }
-      return Optional.of(new QualityGateData(defaultQualityGate.get(), true));
+      QualityGateDto defaultQualityGate = dbClient.qualityGateDao().selectByOrganizationAndUuid(dbSession, organization, organization.getDefaultQualityGateUuid());
+      checkState(defaultQualityGate != null, format("Unable to find the quality gate [%s] for organization [%s]", organization.getDefaultQualityGateUuid(), organization.getUuid()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Invoke method(s) only conditionally. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2629)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 181 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)